components/public-api/gitpod/v1/user.proto

```diff
@@ -0,0 +1,212 @@
+syntax = "proto3";
+
+package gitpod.v1;
+
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/v1";
+
+service UserService {
+  // GetAuthenticatedUser allows to retrieve the current user.
+  rpc GetAuthenticatedUser(GetAuthenticatedUserRequest) returns (GetAuthenticatedUserResponse) {}
+}
+
+message GetAuthenticatedUserRequest {}
+message GetAuthenticatedUserResponse {
+  User user = 1;
+}
+
+message User {
+  // id is a UUID of the user
+  string id = 1;
+
+  // organization_id is the id of the organization this account is owned by.
+  //
+  // +optional if not set, this account is owned by the installation.
+  string organization_id = 2;
+
+  // name is the username
+  string name = 3;
+
+  // avatar_url is a link to the user avatar
+  string avatar_url = 4;
+
+  // created_at is the creation time
+  google.protobuf.Timestamp created_at = 5;
+
+  repeated Identity identities = 6;
+
+  // blocked indicates whether the account is blocked, e.g. due to vialoation of ToS.
+  bool blocked = 7;
+
+  // last_verification_time is the last time this account was verified.
+  //
+  // +optional if not set, this account is not verified.
+  google.protobuf.Timestamp last_verification_time = 8;
+
+  // verification_phone_number is the phone number used for the last verification.
+  string verification_phone_number = 9;
+
+  AdditionalUserData additional_user_data = 10;
+  message AdditionalUserData {
+    // user globol workspace timeout
+    //
+    // +optional
+    string workspace_timeout = 1;
+
+    // control whether to enable the closed timeout of a workspace, i.e. close web ide, disconnect ssh connection
+    //
+    // +optional
+    bool disabled_closed_timeout = 2;
+
+    // +optional
+    EmailNotificationSettings email_notification_settings = 3;
+    message EmailNotificationSettings {
+      // +optional
+      bool allows_changelog_mail = 1;
+      // +optional
+      bool allows_devx_mail = 2;
+      // +optional
+      bool allows_onboarding_mail = 3;
+    }
+
+    // +optional
+    bool feature_preview = 4;
+
+    // +optional
+    IDESettings ide_settings = 5;
```

Comment:
Could we call it `EditoSettings` instead on API level? in docs we don't talk about IDEs but editors

Reply:
done